File: ported/auto/402ada.js
// Auto-translated from Ghidra C by tools/c-to-js/translate.js.
// Source: decompiled/c/402ada.c
// Edit by hand only after diff-test passes — re-running the translator will overwrite.

/** @typedef {import("../runtime/heap.js").Heap} Heap */

import { GetCurrentProcess, GetCurrentThread, GetPriorityClass, GetThreadPriority, SetPriorityClass, SetThreadPriority } from "../runtime/win32.js";
export function FUN_00402ada(heap) {
  let pvVar1 = 0;
  let dwPriorityClass = 0;
  let nPriority = 0;
  if (heap.u32(0x005e912c) != 0) {
    if (heap.u32(0x005e9128) == 0) {
      heap.setU32(0x005e915c, (1) >>> 0);
      pvVar1 = GetCurrentProcess(heap);
      heap.setU32(0x005f2408, (GetPriorityClass(heap, pvVar1)) >>> 0);
      pvVar1 = GetCurrentThread(heap);
      heap.setU32(0x005f1fe8, (GetThreadPriority(heap, pvVar1)) >>> 0);
      dwPriorityClass = 0x100;
      pvVar1 = GetCurrentProcess(heap);
      SetPriorityClass(heap, pvVar1, dwPriorityClass);
      nPriority = 0xf;
      pvVar1 = GetCurrentThread(heap);
      SetThreadPriority(heap, pvVar1, nPriority);
      heap.setU32(0x005e9128, (1) >>> 0);
    } else {
      heap.setU32(0x005e915c, (heap.u32(0x005e915c) + 1) >>> 0);
    }
  }
  return heap.u32(0x005e915c);
}

File: decompiled/c/402ada.c

int FUN_00402ada(void)

{
  HANDLE pvVar1;
  DWORD dwPriorityClass;
  int nPriority;
  
  if (DAT_005e912c != 0) {
    if (DAT_005e9128 == 0) {
      DAT_005e915c = 1;
      pvVar1 = GetCurrentProcess();
      DAT_005f2408 = GetPriorityClass(pvVar1);
      pvVar1 = GetCurrentThread();
      DAT_005f1fe8 = GetThreadPriority(pvVar1);
      dwPriorityClass = 0x100;
      pvVar1 = GetCurrentProcess();
      SetPriorityClass(pvVar1,dwPriorityClass);
      nPriority = 0xf;
      pvVar1 = GetCurrentThread();
      SetThreadPriority(pvVar1,nPriority);
      DAT_005e9128 = 1;
    }
    else {
      DAT_005e915c = DAT_005e915c + 1;
    }
  }
  return DAT_005e915c;
}

